spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -81,570 +85,571 @@
-
-        model.addAttribute("testService", gettingStartedTestService);
-        model.addAttribute("proxiesPerApiKey", proxyURLPerNewApiKey);
-        model.addAttribute("apiKeys", newApiKey);
-
-        // decide about which fragment to return
-        String returnedFragment = (whichFragmentToLoad.equals("table"))
-                // returns the ApiKeyTable Fragment
-                ? "admin/fragments/api-keys/consumer/list :: apiKeyTable"
-                // returns the Url Fragment
-                : "admin/fragments/consumer_wizard/wizard :: gatewayCallURL";
-
-        return returnedFragment;
-    }
-
-    private ServiceApi loadTestServiceForConsumerWizard() {
-        //Getting the test service
-        log.debug("trying to load test service for provider '{}' and uriIdentifier '{}'",
-                gettingStartedTestServiceProvider,
-                gettingStartedTestServiceIdentifier);
-        ServiceApi testServiceApi = serviceApiRepository.findServiceApiByServiceOwnerAndUriIdentifier(
-                gettingStartedTestServiceProvider,
-                gettingStartedTestServiceIdentifier);
-        log.debug("test service loaded: {}", testServiceApi);
-        return testServiceApi;
-    }
+						// temporalFirstTimeFlag = false;
+						mav.setViewName(ADMIN_STARTPAGE);
+
+					} else {
+						// IT IS A CONSUMER USER
+						mav.setViewName(ADMIN_CONSUMER_HOME_VIEW);
+					}
+
+				}
+
+			} else {
+
+				// The user is new for our database therefore we try to retrieve as much user
+				// info is possible from Github
+				OAuth2AuthenticationDetails details = (OAuth2AuthenticationDetails) auth.getDetails();
+
+				RestTemplate restTemplate = new RestTemplate();
+
+				HttpHeaders headers = new HttpHeaders();
+				headers.add("Authorization", "token " + details.getTokenValue());
+				HttpEntity<String> githubRequest = new HttpEntity<String>(headers);
+
+				ResponseEntity<String> userInfoResponse = restTemplate.exchange(GITHUB_API_USER, HttpMethod.GET,
+						githubRequest, String.class);
+				String userInfo = userInfoResponse.getBody();
+				// String userInfo = restTemplate.getForObject(GITHUB_API_USER +
+				// "?access_token=" + details.getTokenValue(), String.class);
+
+				ObjectMapper objectMapper = new ObjectMapper();
+				objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+				Map<String, Object> userMap = objectMapper.readValue(userInfo,
+						new TypeReference<Map<String, Object>>() {
+						});
+				String email = (String) userMap.get("email");
+				if (email == null || email.isEmpty()) {
+					
+					ResponseEntity<String> userEmailsResponse = restTemplate.exchange(GITHUB_API_EMAIL, HttpMethod.GET,
+							githubRequest, String.class);
+					String userEmails = userEmailsResponse.getBody();
+					//String userEmails = restTemplate.getForObject(GITHUB_API_EMAIL + "?access_token=" + details.getTokenValue(), String.class);
```

Comment:
More commented code, please if you will not use it on the future, then remove it

Reply:
well, so far, it is the quick fixed